Mathlib/Analysis/Complex/Hadamard.lean

```diff
@@ -0,0 +1,488 @@
+/-
+Copyright (c) 2023 Xavier Généreux. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Xavier Généreux
+-/
+
+import Mathlib.Analysis.Complex.AbsMax
+import Mathlib.Analysis.SpecialFunctions.Pow.Deriv
+
+/-!
+# Hadamard three-lines Theorem
+
+In this file we present a proof of a special case Hadamard's three-lines Theorem.
+We think this result generalises well by a change of variables.
+
+## Main result
+
+- `abs_le_interp_on_closed_strip` :
+Hadamard three-line theorem on `[0,1]`: If `f` is a bounded function, continuous on `[0,1]`
+and differentiable on `(0,1)`, then for `M(x) := sup ((abs ∘ f) '' (re ⁻¹' {x}))` we have that
+`∀ z ∈ [0,1]` the inequality `|f(z)| ≤ |M(0)^(1-z)| * |M(1)^z|` holds. This can be seen to be
+equivalent to the statement that `log M(x)` is a convex function on `[0,1]`.
+
+## Notation
+
+- `Complex.HadamardThreeLines.strip` : The vertical strip defined by : re ⁻¹' Ioo a b
+
+- `Complex.HadamardThreeLines.closedStrip` : The vertical strip defined by : re ⁻¹' Icc a b
+
+- `Complex.HadamardThreeLines.bddStrip` :
+    The rectangle defined by : re ⁻¹' Ioo a b ∩ im ⁻¹' Ioo (-T) T
+
+- `Complex.HadamardThreeLines.closedBddStrip` :
+    The rectangle defined by : re ⁻¹' Icc a b ∩ im ⁻¹' Icc (-T) T
+
+- `Complex.HadamardThreeLines.sSupAbsIm` :
+    The supremum function on vertical lines defined by : sSup {|f(z)| : z.re = x}
+
+- `Complex.HadamardThreeLines.invInterpStrip` :
+    Inverse of the interpolation between the sSupAbsIm on the edges of the strip.
+
+- `Complex.HadamardThreeLines.F` :
+    Function defined by f times invInterpStrip. Convenient form for proofs.
+
+- `Complex.HadamardThreeLines.F'` :
+    'Easy' function converging to F.
+
+- `Complex.HadamardThreeLines.cocompactStrip` :
+    The filter along the vertical strip re ⁻¹' Ioo a b and |z.im| atTop.
+
+## References
+
+This proof follows loosely the proof found on the wiki page :
+[wiki](https://en.wikipedia.org/wiki/Hadamard_three-lines_theorem)
+
+-/
+
+
+open Set Filter Function Complex
+
+namespace Complex
+namespace HadamardThreeLines
+variable (z : ℂ)
+
+/-- The strip in the complex plane containing all `z ∈ ℂ` such that `z.re ∈ Ioo a b`. -/
+def strip (a: ℝ) (b: ℝ) := re ⁻¹' Ioo a b
```

Comment:
All of these should have type annotations to avoid surprises:
```suggestion
def strip (a : ℝ) (b : ℝ) : ℝ := re ⁻¹' Ioo a b
```
(note also the spacing around colons)

Reply:
Ok will do it right now. Thanks!